### ISSUE TITLE
Add a member function `into_split` for `cedar_policy::Entity`

### ISSUE THREAD

**Issue by 0x00A5** [cedar-policy/cedar#636]
### Category

User level API features/changes

### Describe the feature you'd like to request

Add a new member function, `into_split`, under `Entity` with the following function signature:

```rust
/// Disassemble an `Entity` into its unique ID, attributes and parents.
pub fn into_split(self) -> (EntityUid, HashMap<String, RestrictedExpression>, HashSet<EntityUid>);
```

This function will allow users to mutate an entity's attributes and parents after splitting and then re-assemble back into a `cedar_policy::Entity`.

### Describe alternatives you've considered

None

### Is this something that you'd be interested in working on?

- [X] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by aaronjeline** [comment]
Seems useful, only note is that I think the more idiomatic name would be `into_inner()`

**Comment by khieta** [comment]
Closed by #685
